code-challenges/array-shift/array-shift.test.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const middleInsert = function(arr, value) {
+    if(Array.isArray(arr)) {
+        let middleIndex = Math.ceil(arr.length/2);
+        const stopPoint = arr.length - middleIndex;
+        let newArr = [];
+        for(let i = 0; i < middleIndex; i++) {
+          newArr[i] = arr[i]
+        }
+        newArr[middleIndex] = value;
+        for(let i = middleIndex; i < arr.length; i++) {
+          newArr[i+1] = arr[i];
+        }
+        return newArr;
+    } else {
+        return null;
```

Comment:
I don't think it is useful to return null here. I would likely throw an error here with some sort of useful message. The app is likely to break anyway if you're expecting to receive an array and get null.